Apps/CocoaPods-FCM/src/View/DashboardView.swift

```diff
@@ -1,8 +1,102 @@
 import CioDataPipelines
+import CioMessagingInApp
 import SwiftUI
+import UIKit
 import UserNotifications
 
+// ObservableObject that tracks the height of a subview
+class HeightTracker: ObservableObject {
+    @Published var subviewHeight: CGFloat = 0
+}
+
+// Used for associating the height tracker with InAppMessageView
+private var heightTrackerKey: UInt8 = 0
+private var maxHeight: CGFloat = 0
+
+extension InAppMessageView {
+    var heightTracker: HeightTracker? {
+        get {
+            objc_getAssociatedObject(self, &heightTrackerKey) as? HeightTracker
+        }
+        set {
+            objc_setAssociatedObject(self, &heightTrackerKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+
+    // Tracks and update the height
+    override public func layoutSubviews() {
+        super.layoutSubviews()
+
+        print("Height of tracker \(String(describing: heightTracker?.subviewHeight))")
+
+        // Update maxHeight and notify heightTracker if the frame height changes
+        // The height fluctuates, increasing and then decreasing.
+        // To prevent this behavior, we constrain it to the maximum height achieved by the subview.
+        if frame.height >= maxHeight {
+            maxHeight = frame.height
+            heightTracker?.subviewHeight = frame.height
+        }
+    }
+}
+
+struct InAppMessageViewRepresentable: UIViewRepresentable {
+    var elementId: String
+    @Binding var containerWidth: CGFloat
+    @ObservedObject var heightTracker: HeightTracker
+    func makeUIView(context: Context) -> InAppMessageView {
+        let inlineMessageView = InAppMessageView(elementId: elementId)
+
+        // This is optional. If set, the delegate method `onActionClick`
+        // will receive callbacks.
+        // If not set, the global method `messageActionTaken` will handle the callbacks.
+        inlineMessageView.onActionDelegate = context.coordinator
+        inlineMessageView.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
Could you add a comment for why this is needed?

Reply:
Outdated as moved to SDK